### PR TITLE
feat(tgdump): introduce `--scan-parallel` option

### PR DESCRIPTION
**Pull request by ashigeru**
This pull request adds support for configuring the maximum number of parallel tasks for table scans in the `tgdump` tool via a new `--scan-parallel` option. This allows users to control how many parallel scan operations the database should perform, complementing the existing client-side threading options. The changes include updates to documentation, CLI argument parsing and validation, core model propagation, and test coverage for the new parameter.

**New parameter support and propagation:**

* Added a new CLI option `--scan-parallel` to specify the maximum number of parallel scan tasks for table reads, including validation to ensure non-negative values and documentation updates in both English and Japanese manuals. [[1]](diffhunk://#diff-db60da0c7f268ec3538c2ece3e8f4e038f32d4850bb43566c3454a856643c9b8R95-R104) [[2]](diffhunk://#diff-a5ceebd72c6de881e1f7f0634245879525e5dd81343c5ade32383ea3bc7e914cR109-R112) [[3]](diffhunk://#diff-6faa3bcbf7be06033313f875243983d7a5c225e9834fd0a3b91bf7ab23a50fbbR577-R606)
* Extended the `CommandArgumentSet` class to store and validate the new parameter, including a `ZeroOrMoreValidator` for non-negative integer enforcement. [[1]](diffhunk://#diff-6faa3bcbf7be06033313f875243983d7a5c225e9834fd0a3b91bf7ab23a50fbbR67-R81) [[2]](diffhunk://#diff-6faa3bcbf7be06033313f875243983d7a5c225e9834fd0a3b91bf7ab23a50fbbR204-R205)
* Updated the `TransactionSettings` core model to accept, store, and propagate the `scanParallel` value, ensuring it is included in protocol buffer serialization and equality/hash logic. [[1]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150R22-R25) [[2]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150L60-R67) [[3]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150R106-R121) [[4]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150R140-R141) [[5]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150R160) [[6]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150R195-R205) [[7]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150R236-R242) [[8]](diffhunk://#diff-dea10925be89fdcec3d072d6f4bace3bc21454029655512bec81aff8c1573150L222-R265)
* Modified the main execution flow to pass the new parameter from CLI arguments to transaction settings.

**Testing and utility updates:**

* Added comprehensive unit tests for parsing, validation, and propagation of the `--scan-parallel` option, including edge cases for zero and negative values. [[1]](diffhunk://#diff-7b7f3495caf0a620071399aa0d10e0d027b17b40baaef740c8889ab6ea2ededeR514-R539) [[2]](diffhunk://#diff-271484c34683ec623f886b6a6e8ebbdc7f9fa7c075ca0896109c100664fbaaa4L18-R22) [[3]](diffhunk://#diff-271484c34683ec623f886b6a6e8ebbdc7f9fa7c075ca0896109c100664fbaaa4R37) [[4]](diffhunk://#diff-271484c34683ec623f886b6a6e8ebbdc7f9fa7c075ca0896109c100664fbaaa4R47-R60) [[5]](diffhunk://#diff-271484c34683ec623f886b6a6e8ebbdc7f9fa7c075ca0896109c100664fbaaa4R123-R150)
* Updated argument printing utilities to display the new option for better traceability in logs and diagnostics.

**Documentation and cleanup:**

* Improved documentation to describe the new option and its interaction with threading and output channel limits, and removed obsolete notes about future parallelism options.

These changes provide users with finer control over database-side parallelism for table scans, while ensuring robust validation, clear documentation, and consistent propagation throughout the tool.

### CODE REVIEW

**Comment by ashigeru** [comment]
@akirakw 

* [tgdump の --scan-parallel に関するスモークテスト](https://nautilus-technologies.app.box.com/file/1978385855771) (Box)

**Review by akirakw — APPROVED** [review]
LGTM

File: modules/tgdump/core/src/main/java/com/tsurugidb/tools/tgdump/core/model/TransactionSettings.java [hunk 1]
```diff
@@ -219,11 +254,14 @@ public boolean equals(Object obj) {
...
+        return enableReadAreas == other.enableReadAreas
+            && Objects.equals(label, other.label)
+            && type == other.type
+            && Objects.equals(scanParallel, other.scanParallel);
     }
 
     @Override
     public String toString() {
-        return String.format("TransactionSettings(type=%s, label=%s)", type, label); //$NON-NLS-1$
+        return String.format("TransactionSettings(type=%s, label=%s, scanParallel=%s)", type, label, scanParallel);
```
Comment (hishidama) [inline modules/tgdump/core/src/main/java/com/tsurugidb/tools/tgdump/core/model/TransactionSettings.java:265]: `$NON-NLS-1$` のコメントが消えているが、問題ない？

Comment (ashigeru) [reply modules/tgdump/core/src/main/java/com/tsurugidb/tools/tgdump/core/model/TransactionSettings.java]: fixed: https://github.com/project-tsurugi/tanzawa/pull/53/commits/22f3c2180345dadb8f4a1864ed578a3eb72e177a
